File: C++/프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(string s) {
    vector<int> alphaIndex(26);
    for(int i = 0; i < 26; i++) {
        alphaIndex[i] = 10001;
    }
    
    vector<int> answer(s.size());
    int i = 0;
    for(auto& c : s) {
        answer[i] = max(i - alphaIndex[c - 'a'], -1);
        alphaIndex[c - 'a'] = i++;
    }
    
    return answer;
}

File: Python3/프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    alpha = {chr(i): 10001 for i in range(ord('a'), ord('z') + 1)}
    
    answer = list()
    for i, c in enumerate(s):
        answer.append(max(-1, i - alpha[c]))
        alpha[c] = i
    
    return answer